### PR TITLE
first attempt to properly pass parameters to qphix from input file

### PR DESCRIPTION
**Pull request by kostrzewa**
1) pass parameters to qphix from input file, new "BeginExternalInverter QPHIX"
   structure implemented for this purpose (warning: no correctness checks here)

2) move _initQphix call into invert_eo_qphix

3) add sample file to call qphix Wilson inverter

4) use angled brackets to include qphix headers

caveats: single precision results in a segfault, only the Wilson operator is currently supported

### CODE REVIEW

**Comment by kostrzewa** [comment]
There are some unsafe things done here. For one, the qphix parameters won't exist when building without qphix, so the section in ```read_input.l``` which sets them should be guarded accordingly. We'll get around eventually to fixing these things up.

Is it okay how I call ```_initQphix``` now?

For some reason, single precision fails. However, it seems that everything else works as expected, as long as one parallelises only in two dimensions. In particular, all threading that I tired works correctly.

File: invert_eo.c [hunk 1]
```diff
@@ -102,7 +102,8 @@ int invert_eo(spinor * const Even_new, spinor * const Odd_new,
     return invert_eo_qphix(Even_new, Odd_new, Even, Odd,
                            precision, max_iter,
                            solver_flag, rel_prec,
-                           solver_params, compression);
+                           solver_params,
```
Comment (kostrzewa) [inline invert_eo.c:105]: this was an unwanted change

File: qphix_interface.cpp [hunk 2]
```diff
@@ -128,7 +137,7 @@ int PadXYZ;
 int MinCt;
 int N_simt;
 bool compress12;
-QphixPrec precision;
+QphixPrec qphix_precision;
```
Comment (kostrzewa) [inline qphix_interface.cpp:140]: lots of "precision" going around

File: qphix_interface.cpp [hunk 3]
```diff
@@ -949,16 +960,25 @@ int invert_eo_qphix(spinor * const Even_new,
...
     masterPrintf("# INITIALIZING QPHIX SOLVER\n");
     masterPrintf("# USING SINGLE PRECISION\n");
+    _initQphix(0, nullptr, qphix_input_By, qphix_input_Bz, qphix_input_NCores,
+           qphix_input_Sy, qphix_input_Sz,
+           qphix_input_PadXY, qphix_input_PadXYZ, qphix_input_MinCt,
+           compression, QPHIX_FLOAT_PREC);
+
     if (compress12) {
-      return invert_eo_qphix_helper<float, VECLEN_DP, QPHIX_SOALEN, true>
+      return invert_eo_qphix_helper<float, VECLEN_SP, QPHIX_SOALEN, true>
```
Comment (kostrzewa) [inline qphix_interface.cpp:981]: is this correct? Whatever I do here, I get segfaults

Comment (kostrzewa) [reply qphix_interface.cpp:981]: @plabus: was this the correct thing to do?

Comment (plabus) [reply qphix_interface.cpp:981]: I think we need this as well, but I'm not sure the reordering routines work properly for precisions other than double right now... They will have to be properly templatised, but there is this the issue that the precision used in tmlQCD may be different form the one used in QPhiX (or the interface for that matter).

**Pull request merged**